c/datasetjson.c

```diff
@@ -1719,6 +1720,116 @@ void respondWithHLQNames(HttpResponse *response, MetadataQueryCache *metadataQue
 #endif /* __ZOWE_OS_ZOS */
 }
 
+static int getDSCB(char* datasetName, char* dscb, int bufferSize){
+  if (bufferSize < INDEXED_DSCB){
+    zowelog(NULL, LOG_COMP_DATASETJSON, ZOWE_LOG_WARNING, 
+            "DSCB of size %d is too small, must be at least %d", bufferSize, INDEXED_DSCB);
+    return 1;
+  }
+  Volser volser = {0};
+
+  DatasetName dsn = {0};
+  memcpy(dsn.value, datasetName, DATASET_NAME_LEN);
+
+  int volserSuccess = getVolserForDataset(&dsn, &volser);
+  if(!volserSuccess){
+    int rc = obtainDSCB1(dsn.value, sizeof(dsn.value),
+                           volser.value, sizeof(volser.value),
+                           dscb);
+    if (rc == 0){
+      if (DSCB_TRACE){
+        zowelog(NULL, LOG_COMP_DATASETJSON, ZOWE_LOG_WARNING, "DSCB for %.*s found\n", sizeof(dsn.value), dsn.value);
+        dumpbuffer(dscb,INDEXED_DSCB);
+      }
+    }
+    return 0;
+  }
+  else {
+    return 1;
+  }
+}
+
+void newDatasetMember(HttpResponse* response, char* datasetPath, char* memberName) {
+  char dscb[INDEXED_DSCB] = {0};
+  int bufferSize = sizeof(dscb);
+  if (getDSCB(datasetPath, dscb, bufferSize) != 0) {
+    respondWithJsonError(response, "Error decoding dataset", 400, "Bad Request");
+  }
+  else {
+    if (!isPartionedDataset(dscb)) {
+      respondWithJsonError(response, "Dataset must be PDS/E", 400, "Bad Request");
+    }
+    else {
+      char *overwriteParam = getQueryParam(response->request,"overwrite");
+      int overwrite = !strcmp(overwriteParam, "true") ? TRUE : FALSE;
+      char fullPath[DATASET_MEMBER_MAXLEN + 1] = {0};
+      //concatenates dataset name with member name
+      char *dsName = strtok(datasetPath, " ");
+      char *memName = strtok(memberName, " ");
+      snprintf(fullPath, DATASET_MEMBER_MAXLEN + 1, "//'%s(%s)'", dsName, memName);
+      FILE* memberExists = fopen(fullPath,"r");
```

Comment:
This whole chunk could be better if it was like the following. This isn't tested, it's just a rough outline.

      bool datasetExists = FALSE;
      FILE* datasetMember = fopen(fullPath,"r");
      if (memberExists) {
        datasetExists = TRUE;
      }
      
      int returnCode = fclose()
      if (returnCode != 0) {
        respondWithJsonError();
        return;
      }
      
      if (datasetExists) {
        if (!overwrite) {
          respondWithJsonError();
          return;     
        }
        else {
          createNewDatasetMember(); /* fopen(), fclose() */
        }
      }
      else {
        createNewDatasetMember(); /* fopen(), fclose() */
      }
      
      static int createNewDatasetMember() {
        int returnCode = 0;
        
        returnCode = fopen();
        if (returnCode != 0) {
          zowelog();
          return returnCode;
        }
        
        returnCode = fclose();
        if (returnCode != 0) {
          zowelog();
          return returnCode;
        }
        
        return returnCode;
      }